app/src/main/java/com/example/caregiver/Task_Caregivee.java

```diff
@@ -0,0 +1,195 @@
+package com.example.caregiver;
+
+import android.content.Intent;
+import android.content.SharedPreferences;
+import android.os.Build;
+import android.os.Bundle;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.RequiresApi;
+import androidx.fragment.app.Fragment;
+
+import android.preference.PreferenceManager;
+import android.util.Log;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.AdapterView;
+import android.widget.ListView;
+import android.widget.SimpleAdapter;
+
+import com.google.firebase.database.DataSnapshot;
+import com.google.firebase.database.DatabaseError;
+import com.google.firebase.database.DatabaseReference;
+import com.google.firebase.database.FirebaseDatabase;
+import com.google.firebase.database.ValueEventListener;
+import com.google.gson.Gson;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+/**
+ * A simple {@link Fragment} subclass.
+ * Use the {@link Task_Caregivee#newInstance} factory method to
+ * create an instance of this fragment.
+ */
+public class Task_Caregivee extends Fragment {
+
+    // Global object to store the task list
+    List < Task > taskList = new ArrayList < >();
+
+    // Global view
+    View view;
+
+    /** This represent a single task object */
+    public class Task {
+        String room;/* the room where the task is located */
+        String caregiverId;/* the caregiver who assigned the task */
+        String taskId;/* the task id, auto-generated by Firebase */
+        String taskName;/* task actions (e.g Brush Your Teeth) */
+        String taskNote;/* task notes (Your Brush Is The Red One) */
+        String assignedStatus;/* true = task assigned to caregivee to do, false otherwise*/
+        String completionStatus;/* caregivee complete the tasks (complete), incomplete otherwise */
+
+        // Default constructor
+        public Task() {};
+
+        // Alternative constructor to instantiate the task object.
+        public Task(String room, String caregiverId, String taskId, String taskName,
+                    String taskNote, String assignedStatus, String completionStatus) {
+            this.room = room;
+            this.caregiverId = caregiverId;
+            this.taskId = taskId;
+            this.taskName = taskName;
+            this.taskNote = taskNote;
+            this.completionStatus = completionStatus;
+            this.assignedStatus = assignedStatus;
+        }
+    }
```

Comment:
I wonder if we can leverage the Task class in TaskFragment.java so we don't have to create another one here

Reply:
oh yeah, I think in future we need to maybe create a data model file that handle all the commonly used objects, like User, Task and stuff  like that. I'll add this on my to do list for spring term